bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongLongPairHashMap.java

```diff
@@ -336,9 +399,10 @@ boolean put(long key1, long key2, long value1, long value2, int keyHash, boolean
                     bucket = (bucket + 4) & (table.length - 1);
                 }
             } finally {
-                if (usedBuckets > resizeThreshold) {
+                if (usedBuckets > resizeThresholdUp) {
                     try {
-                        rehash();
+                        // Expand the hashmap
+                        rehash((int) (capacity * (1 + expandFactor)));
```

Comment:
If `expandFactor == 2`, I would expect the size to double each time.

I think this should be:
```suggestion
                        rehash((int) (capacity * expandFactor));
```

Reply:
fixed